torchvision/prototype/transforms/_auto_augment.py

```diff
@@ -26,134 +28,154 @@ def _get_random_item(self, dct: Dict[K, V]) -> Tuple[K, V]:
         key = keys[int(torch.randint(len(keys), ()))]
         return key, dct[key]
 
-    def _apply_transform(self, sample: Any, transform_id: str, magnitude: float) -> Any:
-        def dispatch(
-            image_tensor_kernel: Callable,
-            image_pil_kernel: Callable,
-            input: Any,
-            *args: Any,
-            **kwargs: Any,
-        ) -> Any:
-            if isinstance(input, (features.BoundingBox, features.SegmentationMask)):
-                raise TypeError(f"{type(input).__name__}'s are not supported by {type(self).__name__}()")
-            elif isinstance(input, features.Image):
-                output = image_tensor_kernel(input, *args, **kwargs)
-                return features.Image.new_like(input, output)
-            elif isinstance(input, torch.Tensor):
-                return image_tensor_kernel(input, *args, **kwargs)
-            elif isinstance(input, PIL.Image.Image):
-                return image_pil_kernel(input, *args, **kwargs)
-            else:
-                return input
+    def _query_image(self, sample: Any) -> Union[PIL.Image.Image, torch.Tensor, features.Image]:
+        images = list(query_images(sample))
+        if len(images) > 1:
+            raise TypeError(
+                f"Auto augment transformations are only properly defined for a single image, but found {len(images)}."
+            )
+        return images[0]
 
-        image = query_image(sample)
+    def _parse_fill(self, sample: Any) -> Optional[List[float]]:
+        image = self._query_image(sample)
         num_channels = F.get_image_num_channels(image)
 
         fill = self.fill
         if isinstance(fill, (int, float)):
             fill = [float(fill)] * num_channels
         elif fill is not None:
             fill = [float(f) for f in fill]
+        return fill
+
+    def _dispatch(
+        self,
+        image_tensor_kernel: Callable,
+        image_pil_kernel: Callable,
+        input: Any,
+        *args: Any,
+        **kwargs: Any,
+    ) -> Any:
+        if isinstance(input, (features.BoundingBox, features.SegmentationMask)):
+            raise TypeError(f"{type(input).__name__}'s are not supported by {type(self).__name__}()")
+        elif isinstance(input, features.Image):
+            output = image_tensor_kernel(input, *args, **kwargs)
+            return features.Image.new_like(input, output)
```

Comment:
I'm not sure I understand. Do you mean to not copy meta data from the input image, but rather create a new one?

```suggestion
            return features.Image(output)
```

Reply:
Hmm I just realized that the `image_size` is a property that returns the shape of the underlying data rather than a value cached on the object. So my comment doesn't make sense, you are covered.